activesupport/CHANGELOG.md

```diff
@@ -1,3 +1,23 @@
+*   ActiveSupport::Subscriber#attach_to now accepts an inherit_from: argument. When set to true,
+    it allows a subscriber to receive events for methods defined in the subscriber's ancestor class(es)
+
+    ```ruby
+    class StatsSubscriber < ActiveSupport::Subscriber
+      attach_to :active_record
+
+      def sql(event)
+        Statsd.timing("sql.#{event.payload[:name]}", event.duration)
+      end
+    end
+
+    class CustomStatsSubscriber < StatsSubscriber
+      attach_to :active_record, inherit_all: true
+      # => CustomStatsSubscriber will process events for "sql.active_record" notifications
```

Comment:
This example should probably define a different behavior for `sql` and `StatsSubscriber` should probably have another method. Also because you didn't detach the `StatsSubscriber` I think we will have two different classes listening to the same event so the `Statsd.timing` will happen twice, will not it?

Reply:
Yes, you're right - I was just trying to show how to call the method with the `inherit_all` option, but probably better to flesh out the example more so it actually makes sense!

Reply:
Yeah. I think I'd use our use case of decorating an existing subscriber by modifying a method and detaching the original subscriber.